### PR TITLE
More info about attack sprite and partner following

### PR DESCRIPTION
**Pull request by marius851000**
There is only one attacks sprite loaded at a time.

Also, most of the data structure of wan files.

### CODE REVIEW

File: headers/types/common/enums.h [hunk 1]
```diff
@@ -3649,4 +3649,14 @@ enum monster_gender {
     GENDER_GENDERLESS = 3,
 };
 
+typedef enum wan_source_type {
```
Comment (UsernameFodder) [inline headers/types/common/enums.h:3652]: No typedefs for enums please. You're already using `enum wan_source_type` everywhere else anyway.

```c
enum wan_source_type {
    ...
};
```

File: headers/types/dungeon_mode/dungeon_mode.h [hunk 2]
```diff
@@ -1804,6 +1804,13 @@ struct two_turn_move_and_status {
 };
 ASSERT_SIZE(struct two_turn_move_and_status, 4);
 
+// Store data about the currently loaded attack sprite, including the sprite itself
+struct LoadedAttackSpriteData {
```
Comment (UsernameFodder) [inline headers/types/dungeon_mode/dungeon_mode.h:1808]: The convention of this repo is to use `snake_case` struct tags

File: headers/functions/arm9.h [hunk 3]
```diff
@@ -317,6 +317,9 @@ int LoadWanTableEntryFromPackUseProvidedMemory(struct wan_table* wan_table,
 int ReplaceWanFromBinFile(struct wan_table* wan_table, int wan_id, enum pack_file_id pack_id,
                           uint16_t file_index, bool compressed);
 void DeleteWanTableEntryVeneer(struct wan_table* wan_table, int wan_id);
+bool DoesWanHaveAnimationGroup(struct wan_header* wan_header, uint16_t animation_group_id);
```
Comment (UsernameFodder) [inline headers/functions/arm9.h:320]: Can we shorten this to `WanHasAnimationGroup`?

File: headers/functions/arm9.h [hunk 4]
```diff
@@ -317,6 +317,9 @@ int LoadWanTableEntryFromPackUseProvidedMemory(struct wan_table* wan_table,
...
+void DoesSpriteInWanTableHaveAnimationGroup(uint16_t sprite_id, uint16_t animation_group_id);
```
Comment (UsernameFodder) [inline headers/functions/arm9.h:321]: Can we shorten this to `WanTableSpriteHasAnimationGroup`? Also, the return type should be `bool`?

File: headers/functions/overlay11.h [hunk 5]
```diff
@@ -29,6 +29,10 @@ void SetAnimDataFields2(struct animation* anim, uint32_t flags, uint32_t param_3
 void LoadObjectAnimData(struct animation* anim, int16_t object_id, uint32_t flags);
 void InitAnimDataFromOtherAnimDataVeneer(struct animation* dst, struct animation* src);
 void AnimRelatedFunction(struct animation* anim, undefined4 param_2, undefined4 param_3);
+void AllocAndInitializePartnerFollowDataAndLiveActorList();
```
Comment (UsernameFodder) [inline headers/functions/overlay11.h:32]: Let's shorten `Initialize` to `Init`, per convention of all the other `Init` functions.

Also, a function with no params needs to be explicitly marked `(void)` in C.

File: headers/functions/overlay11.h [hunk 6]
```diff
@@ -29,6 +29,10 @@ void SetAnimDataFields2(struct animation* anim, uint32_t flags, uint32_t param_3
...
+void InitializePartnerFollowDataAndLiveActorList();
```
Comment (UsernameFodder) [inline headers/functions/overlay11.h:33]: Let's shorten `Initialize` to `Init`, per convention of all the other `Init` functions.

Also, a function with no params needs to be explicitly marked `(void)` in C.

File: headers/functions/overlay11.h [hunk 7]
```diff
@@ -29,6 +29,10 @@ void SetAnimDataFields2(struct animation* anim, uint32_t flags, uint32_t param_3
...
+void DeleteLiveActor(short actor_id);
+void InitializePartnerFollowData();
```
Comment (UsernameFodder) [inline headers/functions/overlay11.h:35]: Let's shorten `Initialize` to `Init`, per convention of all the other `Init` functions.

Also, a function with no params needs to be explicitly marked `(void)` in C.

File: headers/functions/overlay29.h [hunk 8]
```diff
@@ -131,6 +131,9 @@ void SetLeaderAction(void);
 void CheckLeaderTile(void);
 void ChangeLeader(void);
 void ResetDamageData(struct damage_data* damage);
+uint16_t SetAndLoadCurrentAttackAnimation(enum pack_file_id pack_id, uint16_t file_index);
+void ClearLoadedAttackSprite();
```
Comment (UsernameFodder) [inline headers/functions/overlay29.h:135]: A function with no params needs to be explicitly marked `(void)` in C.

File: headers/functions/overlay29.h [hunk 9]
```diff
@@ -131,6 +131,9 @@ void SetLeaderAction(void);
...
+uint16_t GetLoadedAttackSpriteId();
```
Comment (UsernameFodder) [inline headers/functions/overlay29.h:136]: A function with no params needs to be explicitly marked `(void)` in C.

File: headers/types/files/wan.h [hunk 10]
```diff
@@ -1,14 +1,99 @@
 #ifndef HEADERS_TYPES_FILES_WAN_H_
 #define HEADERS_TYPES_FILES_WAN_H_
 
-// For now, there is only the top-level header for WAN files.
+#include "../common/generic.h"
 
-// Directly loaded from memory as they are stored. Names comes from the pmd_wan project.
-#pragma pack(push, 1)
+enum wan_sprite_type {
+    PropsUI = 0, // for object and UI objects,
```
Comment (UsernameFodder) [inline headers/types/files/wan.h:7]: Enum variant names are global to the whole program. Can we make all these more specific? They should also be `SCREAMING_SNAKE_CASE` per convention. Perhaps `WAN_SPRITE_PROPS_UI`, etc.?

File: headers/types/files/wan.h [hunk 11]
```diff
@@ -1,14 +1,99 @@
...
+    Chara = 1,   // for monster sprites,
+    Unk2 = 2,
+    Unk3 = 3
+};
+
+#pragma pack(push, 2)
+ENUM_16_BIT(wan_sprite_type);
+#pragma pack(pop)
+
+struct rgba8 {
```
Comment (UsernameFodder) [inline headers/types/files/wan.h:17]: It seems like we could just reuse `struct rgb` from `file_io.h` and rename the `unused` field to `alpha`, instead of creating a new struct.

File: headers/functions/overlay11.h [hunk 12]
```diff
@@ -29,6 +29,10 @@ void SetAnimDataFields2(struct animation* anim, uint32_t flags, uint32_t param_3
 void LoadObjectAnimData(struct animation* anim, int16_t object_id, uint32_t flags);
 void InitAnimDataFromOtherAnimDataVeneer(struct animation* dst, struct animation* src);
 void AnimRelatedFunction(struct animation* anim, undefined4 param_2, undefined4 param_3);
+void AllocAndInitializePartnerFollowDataAndLiveActorList();
+void InitializePartnerFollowDataAndLiveActorList();
+void DeleteLiveActor(short actor_id);
```
Comment (UsernameFodder) [inline headers/functions/overlay11.h:34]: Convention is to use `int16_t` rather than `short` for explicitness

File: symbols/overlay29.yml [hunk 13]
```diff
@@ -1389,6 +1389,29 @@ overlay29:
         Zeroes the damage data struct, which is output by the damage calculation function.
         
         r0: damage data pointer
+    - name: FreeLoadedAttackSpriteAndMore
+      address:
+        EU: 0x22F78C8
+      description: Free among other another data structure the attack sprite storage area/data
```
Comment (UsernameFodder) [inline symbols/overlay29.yml:1395]: When a function has no inputs or outputs, convention is to put `No params." where the parameter list would normally go. In contrast, not saying anything is used to mean that we don't know the inputs or outputs.

**Review by UsernameFodder — CHANGES_REQUESTED** [review]

File: headers/functions/arm9.h (hunk 3 above)
Comment (UsernameFodder) [reply headers/functions/arm9.h:320]: I don't think it would? "Has" in English typically refers to "the current state of things". It doesn't imply an action like some other words might (e.g., "set", "give").

I don't have a strong preference though.

Comment (marius851000) [reply headers/functions/arm9.h:320]: Indeed. After reviewing this later, I think you are right.

File: headers/types/files/wan.h (hunk 11 above)
Comment (marius851000) [reply headers/types/files/wan.h:17]: I’ll rename this struct to RGBA and comment that alpha is sometimes unused then.

**Comment by marius851000** [comment]
I haven’t yet pushed my local (unfinished) changes. Will do that later today hopefully.

File: symbols/overlay29.yml [hunk 14]
```diff
@@ -7402,6 +7425,24 @@ overlay29:
...
+        type: uint16_t
+    - name: LOADED_ATTACK_SPRITE_PACK_ID
+      address:
+        EU: 0x23541AE
+      length:
+        EU: 0x2
+      description: |-
+        [Runtime] The pack id of the currently loaded attack sprite. Should correspond to the id of m_attack.bin
+        
+        type: enum pack_file_id
```
[the remark this answers is not in the document]
Comment (marius851000) [reply symbols/overlay29.yml:7445]: Yet. Thought they weren’t present in overlay29, but checked again, and these values is in the range.

**Comment by marius851000** [comment]
@UsernameFodder changes has been taken into account and the changes has been rebased.

File: headers/data/ram.h [hunk 15]
```diff
@@ -30,6 +30,7 @@ extern struct team_member_table TEAM_MEMBER_TABLE;
 extern uint32_t FRAMES_SINCE_LAUNCH_TIMES_THREE;
 extern struct sentry_duty SENTRY_DUTY_STRUCT;
 extern bool TURNING_ON_THE_SPOT_FLAG;
+extern struct loaded_attack_sprite_data* LOADED_ATTACK_SPRITE_DATA;
```
Comment (UsernameFodder) [inline headers/data/ram.h:33]: This struct type isn't define anywhere. Did you mean to define it somewhere?

**Review by UsernameFodder — CHANGES_REQUESTED** [review]
Thanks, just a few minor questions left, then we'll be good. Made a couple minor tweaks, mainly formatting and grammar, plus filling in some forgotten headers.

File: symbols/arm9.yml [hunk 16]
```diff
@@ -5708,9 +5736,9 @@ arm9:
           - 0x2052A44
           - 0x2052A60
       description: |-
-        Note: unverified, ported from Irdkwia's notes
+        Return the index for the three wan sprite of this monster
```
[the remark this answers is not in the document]
Comment (marius851000) [reply symbols/arm9.yml:5739]: There is three sprites, but the id is the same for the three of them. I’ll update the description.

File: headers/data/ram.h (hunk 15 above)
Comment (marius851000) [reply headers/data/ram.h:33]: The structure seems to be correctly defined inside `dungeon_mode.h`. Or am I missing something.

**Comment by marius851000** [comment]
rebased to fix conflict

File: headers/data/ram.h (hunk 15 above)
Comment (UsernameFodder) [reply headers/data/ram.h:33]: You're right, not sure how I missed that.

**Review by UsernameFodder — APPROVED** [review]
Looks good. Thanks!
